### ISSUE TITLE
Static resource not recognised on a brand new scaffolded Xamarin Forms app

### ISSUE THREAD

**Issue by eddieyanez** [mfractor/mfractor-feedback#309]
Once the solution is created, the App.xaml file contains the code listed below.

On line 18 with <Setter Property="BackgroundColor" Value="{StaticResource Primary}" /> the word Primary has squiggly red lines with the tooltip error: "A static resource named 'Primary' is not defined in this document, the application resources or any referenced resource dictionaries". However, Primary is defined on line 10 with <Color x:Key="Primary">#2196F3</Color>.

_**Edit:**
This seems to be an issue if the key name starts with an uppercase character (i.e. 'Primary'), but not when the name starts with a lowercase character (i.e. 'primary'). Is this a bug, or is the tool telling me that resources should be named using camel case?_

<img width="1046" alt="Screenshot 2020-11-29 at 07 10 46" src="https://user-images.githubusercontent.com/1080027/100535771-c7a24500-3213-11eb-92a2-f37268a5b32a.png">

```xml
<?xml version="1.0" encoding="utf-8" ?>
<Application xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="SwitchedApp.Mobile.App">
    <!--
        Define global resources and styles here, that apply to all pages in your app. 
    -->
    <Application.Resources>
        <ResourceDictionary>
            <Color x:Key="Primary">#2196F3</Color>
            <Style TargetType="Button">
                <Setter Property="TextColor" Value="White"></Setter>
                <Setter Property="VisualStateManager.VisualStateGroups">
                    <VisualStateGroupList>
                        <VisualStateGroup x:Name="CommonStates">
                            <VisualState x:Name="Normal">
                                <VisualState.Setters>
                                    <Setter Property="BackgroundColor" Value="{StaticResource Primary}" />
                                </VisualState.Setters>
                            </VisualState>
                            <VisualState x:Name="Disabled">
                                <VisualState.Setters>
                                    <Setter Property="BackgroundColor" Value="#332196F3" />
                                </VisualState.Setters>
                            </VisualState>
                        </VisualStateGroup>
                    </VisualStateGroupList>
                </Setter>
            </Style>
        </ResourceDictionary>        
    </Application.Resources>
</Application>
```

##Installation Information

```
--Platform Information--
Operating System: Mac
OS: Unix
OS Service Pack: 
OS Version: Unix 20.1.0.0
--MFractor Information--
Product Name: MFractor
Product Version: 4.4.2.0
Product Variant: Visual Studio Mac
Product Variant Version: 8.8.2.41
Product SKU: MFRACTOR-VS-MAC
Commit SHA: dea08f61041ba50eb9aa655b099a0addca130e68
Revision: 4675
Branch: master
Date: 29/11/2020 4:45:45 AM (UTC)
Agent: Local
Installed extensions:	'MFractor (4.4.2)', 'Refactoring Support (8.8.2)', 'JavaScript Debugger (8.8.2)', 'Azure Functions development (8.8.2)', 'Android development (8.8)', 'Web Project Support (8.8.2)', 'Visual Designer Support (8.8.2)', 'Test Runner (8.8.2)', 'Subversion core engine (8.8.2)', 'ILAsm Language Binding (8.8.2)', 'NuGet Packaging (8.8.2)', 'GTK# Visual Designer (8.8.2)', 'MonoDevelop Assembly Browser (8.8.2)', 'Git support (8.8.2)', 'CSharp Language Binding (8.8.2)', 'Project Web References (8.8.2)', 'Xamarin Test Cloud (8.8)', 'Xamarin Test Cloud - NUnit (8.8)', 'Android Deploy (16.8.0.507)', 'Version Control Support (8.8.2)', 'NUnit support (8.8.2)', 'Debugger support for MonoDevelop (8.8.2)', 'Mono Soft Debugger Support (8.8.2)', 'Xamarin Ide Identity (8.8.2)', 'Xamarin Hot Reload (1.0.0.4127551)', 'Text Templating (8.8.2)', 'NuGet Package Management (8.8.2)', 'Azure Support (8.8.2)', 'MonoDevelop Text Editor (8.8.2)', 'Xamarin Designer (16.8.0.507)', 'Web Editors (8.3)', 'Xamarin Terminal (8.8.2)', '.NET Core Support (8.8.2)', 'Xamarin Addins Core (8.8)', 'Container Tools (1.0.20200521.1)', 'Extension Developer Tools (8.8.2)', 'ConnectedServices (8.8.2)', 'Xamarin.Forms Previewer (16.8.0.507)', 'Android Designer (16.8.0.507)', 'Xamarin.Mac development (8.8)', 'Performance Diagnostics of IDE (8.8.2)', 'Unit Testing core support (8.8.2)', 'ChangeLog Add-in (8.8.2)', 'JavaScript and TypeScript (8.3)', 'Razor Language Services (8.0.20201102.3)', 'Xamarin Ide Extensions for Mac (8.8.2)', 'VisualFSharp (8.8.2)', 'Xamarin Ide Extensions (8.8.2)', 'Regex Toolkit (8.8.2)', 'MonoDevelop Mac Platform Support (8.8.2)', 'MonoDevelop Text Editor (8.8.2)', 'VsCode Debug Protocol support for MonoDevelop (8.8.2)', 'Deployment Services Core (8.8.2)', 'Forms Project Support (8.8)', 'Mac Development (8.8)', 'XamlLanguageService (16.8.0.507)', 'iOS development (8.8)', 'Visual Studio for Mac Tools for Unity (2.8.3.0)', 'Mono Soft Debugger Support for ASP.NET (8.8.2)', 'MonoDevelop Source Editor (8.8.2)', 'Gettext Translations Support (8.8.2)', 'Azure Mobile App Service (8.8)', 'MonoDevelop Hex Editor (8.8.2)', 'GDB Debugger Support (8.8.2)', 'Makefile generation (8.8.2)', 'iOS Designer (16.8.0.507)', 'Subversion support (8.8.2)', '.Net Core Debugger (8.8.2)', 'XML Editor (8.8.2)', 'VB.NET Language Binding (8.8.2)', 'ASP.NET Core Support (8.8.2)', 'Azure DevOps Authentication Support (8.8.2)'
```

**Comment by matthewrdev** [comment]
@eddieyanez From what you've described, it looks that MFractor is failing to index projects/solution on creation. I'll investigate.

To address your edit, MFractor is not validating the casing of your resources. I suspect that changing the case from upper to lower triggers MFractors indexer to run and it registers the static resources for validation.

**Comment by matthewrdev** [comment]
@eddieyanez This has been fixed and will be available in MFractor 4.4.4.

**Comment by eddieyanez** [comment]
Thanks that seems to have solved it.
